planning/scenario_planning/common/motion_velocity_optimizer/src/motion_velocity_optimizer.cpp

```diff
@@ -14,177 +14,204 @@
   } else if (p.algorithm_type == "Linf") {
     LinfPseudoJerkOptimizer::OptimizerParam param;
     param.max_accel = p.max_accel;
     param.min_decel = p.min_decel;
-    pnh_.param<double>("pseudo_jerk_weight", param.pseudo_jerk_weight, 200.0);
-    pnh_.param<double>("over_v_weight", param.over_v_weight, 100000.0);
-    pnh_.param<double>("over_a_weight", param.over_a_weight, 5000.0);
-    optimizer_ = boost::make_shared<LinfPseudoJerkOptimizer>(param);
+    param.pseudo_jerk_weight = declare_parameter("pseudo_jerk_weight", 200.0);
+    param.over_v_weight = declare_parameter("over_v_weight", 100000.0);
+    param.over_a_weight = declare_parameter("over_a_weight", 5000.0);
+    optimizer_ = std::make_shared<LinfPseudoJerkOptimizer>(param);
   } else {
-    ROS_ERROR("unknown velocity optimizer.");
+    RCLCPP_ERROR(get_logger(), "unknown velocity optimizer.");
   }
 
-  /* dynamic reconfigure */
-  dynamic_reconfigure::Server<
-    motion_velocity_optimizer::MotionVelocityOptimizerConfig>::CallbackType dyncon_f =
-    boost::bind(&MotionVelocityOptimizer::dynamicRecofCallback, this, _1, _2);
-  dyncon_server_.setCallback(dyncon_f);
-
   /* debug */
-  debug_closest_velocity_ = pnh_.advertise<std_msgs::Float32>("closest_velocity", 1);
-  debug_closest_acc_ = pnh_.advertise<std_msgs::Float32>("closest_acceleration", 1);
-  pub_trajectory_raw_ =
-    pnh_.advertise<autoware_planning_msgs::Trajectory>("debug/trajectory_raw", 1);
-  pub_trajectory_vel_lim_ = pnh_.advertise<autoware_planning_msgs::Trajectory>(
-    "debug/trajectory_external_velocity_limitted", 1);
-  pub_trajectory_latcc_filtered_ =
-    pnh_.advertise<autoware_planning_msgs::Trajectory>("debug/trajectory_lateral_acc_filtered", 1);
-  pub_trajectory_resampled_ =
-    pnh_.advertise<autoware_planning_msgs::Trajectory>("debug/trajectory_time_resampled", 1);
-
-  const double timer_dt = 0.1;
-  timer_ = nh_.createTimer(ros::Duration(timer_dt), &MotionVelocityOptimizer::timerCallback, this);
+  debug_closest_velocity_ =
+    create_publisher<autoware_debug_msgs::msg::Float32Stamped>("closest_velocity", rclcpp::QoS{1});
+  debug_closest_acc_ = create_publisher<autoware_debug_msgs::msg::Float32Stamped>(
+    "closest_acceleration", rclcpp::QoS{1});
+  pub_trajectory_raw_ = create_publisher<autoware_planning_msgs::msg::Trajectory>(
+    "debug/trajectory_raw", rclcpp::QoS{1});
+  pub_trajectory_vel_lim_ = create_publisher<autoware_planning_msgs::msg::Trajectory>(
+    "debug/trajectory_external_velocity_limitted", rclcpp::QoS{1});
+  pub_trajectory_latcc_filtered_ = create_publisher<autoware_planning_msgs::msg::Trajectory>(
+    "debug/trajectory_lateral_acc_filtered", rclcpp::QoS{1});
+  pub_trajectory_resampled_ = create_publisher<autoware_planning_msgs::msg::Trajectory>(
+    "debug/trajectory_time_resampled", rclcpp::QoS{1});
+
+  /* timer */
+  {
+    external_velocity_limit_update_rate_ = declare_parameter("over_a_weight", 10.0);
```

Comment:
only occasion I see where the declared parameter name doesn't match the member variable. Is it on purpose?

Reply:
It's not on purpose. Thanks for letting me know. I'll use your MACRO.

Reply:
Done